src/js/gamestate.js

```diff
@@ -197,18 +214,71 @@ class GameState {
       this.draw(move.player, move.card)
     } else if (move.op == "tickTime") {
       // the server sends the current time for the client to display
-      this.currentGameSecond = move.time
-      window.client.forceUpdate()
+      if (!this.gameTime) {
+        this.startTime = move.gameTime
+        this.gameTime = move.gameTime
+        this.localTimeLoop = setInterval(() => {
+          this.tickLocalTime()
+        }, CLOCK_CYCLE_MS)
+      }
+      this.gameTime = move.gameTime
     } else {
       console.log("ignoring op: " + move.op)
       return false
     }
-
-    this.history.push(move)
+    
+    // client doesn't exist in tests
+    if (window.client) {
+      window.client.forceUpdate()
+    }
+    if (move.op != "tickTime") {
+      this.history.push(move)
+    }
     return true
   }
 
-  // log all moves in the game so far
+  // Tick time locally, and act based on last gameTime sent by server
+  tickLocalTime() {
+    // set currentGameSecond to 10 through 1 based on what second it is in the round
+    this.currentGameSecond = Math.floor(10 - ((this.gameTime - this.startTime) % 10000)/1000)
+    
+    // loop over all cards in play to see if they should act
+    for (let p of this.players) {
+      for (let card of p.board) {
+        this.warmUpAndAttack(card)
+      }
+    }
+    window.client.forceUpdate()
+  }
+
+  // Update the card's opacity and potentially attack
+  warmUpAndAttack(card) {
+    // only creatures attack
+    if (!card.attackRate) {
+      return
```

Comment:
Yeah, because creation time is being set client side, and gameTime is created serverside.

I'll move creationTime to the server.